src/modules/rppt_tensor_effects_augmentations.cpp

```diff
@@ -1148,6 +1148,99 @@ RppStatus rppt_pixelate_host(RppPtr_t srcPtr,
     return RPP_SUCCESS;
 }
 
+/******************** fog ********************/
+
+RppStatus rppt_fog_host(RppPtr_t srcPtr,
+                        RpptDescPtr srcDescPtr,
+                        RppPtr_t dstPtr,
+                        RpptDescPtr dstDescPtr,
+                        Rpp32f *gammaTensor,
+                        RpptROIPtr roiTensorPtrSrc,
+                        RpptRoiType roiType,
+                        rppHandle_t rppHandle)
+{
+    RppLayoutParams layoutParams = get_layout_params(srcDescPtr->layout, srcDescPtr->c);
+
+    RpptDesc fogMaskSrcDesc; 
+    RpptDescPtr fogMaskSrcDescPtr = &fogMaskSrcDesc; // Pointer to original fog mask source descriptor
+    set_fog_mask_descriptor(fogMaskSrcDescPtr, 2, FOG_MAX_HEIGHT, FOG_MAX_WIDTH, 1);
+
+    RpptDesc fogMaskDstDesc;
+    RpptDescPtr fogMaskDstDescPtr = &fogMaskDstDesc;  // Pointer to fog resized mask destination descriptor
+    set_fog_mask_descriptor(fogMaskDstDescPtr, 2, srcDescPtr->h, srcDescPtr->w, 1);
+
+    RpptImagePatchPtr internalDstImgSizes = reinterpret_cast<RpptImagePatch *> (rpp::deref(rppHandle).GetInitHandle()->mem.mcpu.scratchBufferHost);
+    RpptROI *internalRoiTensorPtrSrc = reinterpret_cast<RpptROI *>(internalDstImgSizes + 2);
+    for(Rpp32s i = 0; i < 2; i++)
+    {
+        internalDstImgSizes[i] = {srcDescPtr->w, srcDescPtr->h};
+        internalRoiTensorPtrSrc[i].xywhROI = {0, 0, 1920, 1080};
+    }
+    RpptInterpolationType interpolationType = RpptInterpolationType::NEAREST_NEIGHBOR;
+    Rpp32f *fogAlphaMaskPtr = reinterpret_cast<Rpp32f *>(internalRoiTensorPtrSrc + 2);
+    Rpp32f *fogIntensityMaskPtr = fogAlphaMaskPtr + (srcDescPtr->h * srcDescPtr->w);
+    rppt_resize_host(&fogMask_1920_1080[0], fogMaskSrcDescPtr, fogAlphaMaskPtr, fogMaskDstDescPtr, internalDstImgSizes, interpolationType, internalRoiTensorPtrSrc, roiType, rppHandle);
```

Comment:
Add a comment here mentioning that
we need to resize the mask to the max size present in the batch